teams/vsp/teams/product-dev-support/guidelines-for-reviewing-code.md

```diff
@@ -0,0 +1,81 @@
+
+### Examples of things that should block approval
+
+*   The pull request is too large (i.e. more than 300 lines)
+*   The changes excessively circumvent rules enforced by CI (e.g. disabling linting, writing unit tests that obviously don’t test anything) 
+    *   _Note: there are legitimate reasons to disable linting rules, but use your best judgement. Generally if it’s not a rule that’s been skipped somewhere else in the repo, it’s probably not something we should approve. _
+*   The changes impact shared components / helpers but the PR does not document adequate regression testing was performed
+    *   Changes to /platform or /site folders (especially to the forms system) should be made by or closely supervised by the Tools and Improvements team 
+*   Looping / control structures that obviously could result in unresponsive UIs (e.g. using a `do-while` loop that mutates its terminating condition)
+*   Adding excessive or large static assets to the repo
+    *   We don’t have any image optimization / compression in the build process
+*   Adding files that should be somewhere else (e.g. HTML files should be in the Drupal CMS or the `vagov-content` repo) 
+
+### Examples of things that should not block approval
+
+These are things you can optionally look at, but aren't required to enforce outside of `platform/` and `site/`. We generally offer recommendations, but don't block merging, for these kinds of issues:
+
+*   Poor code style / non-idiomatic JS usage 
+    *   _i.e. using a `for` loop instead of a `.map` _
+    *   _i.e. using conditionals in a loop instead of `.filter` _
+*   Inconsistency with our own style (for other repositories)
+    *   _I.e. ensuring the general approach to GIBCT is maintained is okay. Even if that style doesn’t match our other repositories, it is internally consistent _
+*   The change doesn’t work as intended 
+    *   _Generally we rely on the app teams' peer review / QA to validate this _
+
+### Final words
+
+Use your best judgement.
+
+We aren't responsible for the long term maintainability of app code, but we should endeavor to make it easy to write good, clear code.
+
+
+## Time-boxing
+
+**Each VSP engineer should spend no more than one hour/day performing PR reviews.**
+
+Ideally, this should be a time-boxed activity so that reviews aren’t substantially interfering with other sprint objectives.
+
+### Goals
+
+*   PRs sent to the VSP team are completed in the expected time windows (see [code review guidelines](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md) for VFS teams)
+    *   _An engineer from the VSP team will aim to review within one business day after the pull request has been reviewed by your team and you have been marked it Ready for review. Each additional round of changes will be reviewed within 24 hours._
```

Comment:
Are we formally requesting people to mark it with a tag `Ready for review`? Or move it out of draft status (and/or remove the WIP from the title). I'm amenable to whichever but want to clearly let folks know what the trigger(s) for a review are during onboarding

Reply:
I think we should update it to reflect draft -> pr process. tagging the review group in a comment was a fallback we used when people didn't follow that process but now i just add the `needs peer approval` tag to those PRs.

Reply:
We already tell people (multiple times and multiple places, like the presentations we use/send them) about draft and WIP so education is already in place. Agree we can just update this to reflect what we're already telling people